tests/rbd/test_performance_immutable_cache.py

```diff
@@ -0,0 +1,280 @@
+    real_time_2 = calculate_time(second_read[1])
+
+    # Compare the times
+    time_difference = (
+        real_time_2 - real_time_1
+        if real_time_2 >= real_time_1
+        else real_time_1 - real_time_2
+    )
+
+    log.info(f"First read of cloned image: {real_time_1}")
+    log.info(f"Second read of cloned image: {real_time_2}")
+
+    return time_difference
+
+
+def test_immutable_cache_performance(rbd_obj, **kw):
+    """
+    Test the performance of Immutable Object Cache with and without cache.
+
+    Args:
+        rbd_obj (object): RBD object.
+        kw (dict): Test data.
+
+    Raises:
+        Exception: If any test step fails.
+    """
+    ceph_cluster = kw.get("ceph_cluster")
+    client = ceph_cluster.get_nodes(role="client")[0]
+
+    try:
+        # Configure Immutable Object Cache
+        if configure_immutable_cache(rbd_obj, client, "user1"):
+            log.error("Immutable object cache configuration failed")
+            return 1
+        log.info("Immutable object cache configuration completed successfully.")
+
+        mount_path_with_cache = f"/tmp/mnt_{random_string(len=5)}"
+        mount_path_without_cache = f"/tmp/mnt_{random_string(len=5)}"
+
+        for pool_type in rbd_obj.get("pool_types"):
+            rbd_config = kw.get("config", {}).get(pool_type, {})
+            multi_pool_config = getdict(rbd_config)
+
+            for pool, pool_config in multi_pool_config.items():
+                multi_image_config = getdict(pool_config)
+
+                for image in multi_image_config.keys():
+                    # Perform I/O operations with cache enabled
+                    time_with_cache = perform_clone_io(
+                        client=client,
+                        rbd_obj=rbd_obj,
+                        pool=pool,
+                        image=image,
+                        io_type="write",
+                        mount_path=mount_path_with_cache,
+                        **kw,
+                    )
+
+                    # Removing immutable cache
+                    client.exec_command(cmd=f"rm -rf {immutable_cache_path}", sudo=True)
+
+                    image = image + random_string(len=3)
+
+                    size = kw.get("size", "4G")
+                    img_cmd = f"rbd create {image} --size {size} --pool {pool}"
+                    client.exec_command(sudo=True, cmd=img_cmd)
+
+                    # Perform I/O operations without cache
+                    time_without_cache = perform_clone_io(
+                        client=client,
+                        rbd_obj=rbd_obj,
+                        pool=pool,
+                        image=image,
+                        io_type="write",
+                        mount_path=mount_path_without_cache,
+                        **kw,
+                    )
+
+                    # Check performance comparison
+                    compare_performance(time_with_cache, time_without_cache)
```

Comment:
This wouldn't be time_with_cache and time_without_cache, the better names here would be time_difference_with_cache and time_difference_without_cache. Because basically you're comparing the time difference is large when cache is enabled and its small when cache is not enabled. You are not comparing time taken but you're comparing time difference.

Reply:
done.